lib/chartjs/chart_helpers.rb

```diff
@@ -49,16 +53,9 @@ def chart(klass, data, options)
 
             #{legend if generate_legend}
           };
-          /* W3C standard */
-          if (window.addEventListener) {
-            window.addEventListener("load", initChart, false);
-            document.addEventListener("page:load", initChart, false);
-          }
-          /* IE */
-          else if (window.attachEvent) {
-            window.attachEvent("onload", initChart);
-            document.attachEvent("page:load", initChart);
-          }
+          
+          initChart();
```

Comment:
Please could you try adding `document.addEventListener("ajax:success", initChart, false);` and/or the IE version, and let me know whether that gets thinks working on AJAX calls?